tensorflow/contrib/ffmpeg/default/ffmpeg_lib.cc

```diff
@@ -198,8 +217,76 @@ string BuildWavFile(int32 samples_per_second, int32 channel_count,
   return data;
 }
 
+Status ReadInfoFile(const string& filename, uint32* width, uint32* height,
+                    uint32* frames) {
+  string data;
+  ReadFileToString(Env::Default(), filename, &data);
+  bool in_output = false;
+  bool in_mapping = false;
+  uint32 frames_value = 0;
+  uint32 height_value = 0;
+  uint32 width_value = 0;
+  for (const string& line : str_util::Split(data, '\n')) {
+    if (!in_output && line.find("Output #") == 0) {
```

Comment:
Can we maybe restructure this as:
```
if (MaybeParseOutput(...)) continue;
if (MaybeParseStreamMapping(...)) continue;
...
```
(If you can put `continue` it might better.)

Reply:
Done.